accepted/color-4-new-spaces.md

```diff
@@ -1258,88 +1258,79 @@ a number relative to the range `[0,max]` without clamping.
 
 * If `number` has units other than `%`, throw an error.
 
-* If `number` has the unit `%`, set `number` to `number * max / 100%`.
+* Otherwise, if `number` has the unit `%`, return `number * max / 100%`.
 
-* Return `number`.
+* Otherwise, return `number`.
 
 ### Validating a Color Channel
 
 [validating]: #validating-a-color-channel
 
-This process accepts a SassScript value `channel` to validate, a [known color
-space] `space` to validate against, and the `key` name of the channel. It
+This process accepts a SassScript value `value` to validate, a [known color
+space] `space` to validate against, and `channel` a channel in `space`. It
 throws an error if the channel is invalid for the color space, or returns a
```

Comment:
this error is not described in the algorithm. As the order of errors is observable (by seeing which error you get when passing 2 invalid values), I think it should.

Reply:
It throws an error if the number's value is `NaN`. There aren't any other circumstances where a number is fully invalid for a channel value. Although come to think of it, this should include infinite numbers as well.

Reply:
Actually, rereading the Color 4 spec, it looks like it actually _preserves_ degenerate channel values and just lets them do whatever float stuff they're going to do, so we should do the same and avoid ever throwing an error here.